.circleci/config.yml

```diff
@@ -0,0 +1,166 @@
+version: 2.1
+jobs:
+  build:
+    docker:
+      - image: amd64/centos:7
```

Comment:
@trunkmaster if you look at this line, you will see that we're building on top of CentOS 7.

Are you saying that you are interested in building CentoS 7 directly from the sources using CircleCI?

As long as we can build a Linux distribution in a Docker container, I think it should be fine. I can't think of why it wouldn't be possible. Most modern CI solutions are just scripts running in Docker containers.

If it ends up not being possible, it's also very easy to convert the script into another platform or format.

Reply:
> @trunkmaster if you look at this line, you will see that we're building on top of CentOS 7.

Good.

> Are you saying that you are interested in building CentoS 7 directly from the sources using CircleCI?

Yes. I think that's why CI was invented. Ideally CircleCI should build all project's parts from scratch (including libdispatch, GNUstep libraries).

> As long as we can build a Linux distribution in a Docker container, I think it should be fine. I can't think of why it wouldn't be possible. Most modern CI solutions are just scripts running in Docker containers.

Thanks, now it's clear to me.

> If it ends up not being possible, it's also very easy to convert the script into another platform or format.

CircliCI is OK. Did you test this config file?

Reply:
Yes, if you click the URL in my first post, you can see the build pass.

ALso, if you click this link, you can see a "green checkmark" on each commit where it passed, and a "red X" on each commit where it failed: https://github.com/enzuru/nextspace/commits/circleci-setup-2